liesel/model/model.py

```diff
@@ -955,6 +976,74 @@ def set_seed(self, seed: jax.random.KeyArray) -> Model:
 
         return self
 
+    def simulate(self, seed: jax.random.KeyArray, skip: Iterable[str] = ()) -> Model:
+        """
+        Updates the model state simulating from the probability distributions in the
+        model using a provided random seed, optionally skipping specified nodes.
+
+        Parameters
+        ----------
+        seed
+            The seed is split and distributed to the distribution nodes in the model.
+            Must be a ``KeyArray``, i.e. an array of shape (2,) and dtype ``uint32``.
+            See :mod:`jax.random` for more details.
+        skip
+            The names of the nodes or variables to be excluded from the simulation. \
+            By default, no nodes or variables are skipped.
+
+        Returns
+        -------
+        The model instance itself after updating its state with the simulated values.
+
+        Raises
+        ------
+        AttributeError
+            If the value of the :attr:`.Dist.at` node of a distribution node cannot be
+            set.
+
+        Notes
+        -----
+        The simulation is based on the shapes of the current values of the
+        :attr:`.Dist.at` nodes of the distribution nodes. If the :attr:`.Dist.at` node
+        of a distribution node is a :Class:`.VarValue` node, the value of its input is
+        updated.
+        """
+        dists = [
+            node
+            for node in self._simulation_nodes
+            if isinstance(node, Dist)
+            and node.at is not None
+            and node.name not in skip
+            and node.at.name not in skip
+            and (node.var is not None and node.var.name not in skip)
+        ]
+
+        seeds = jax.random.split(seed, len(dists))
+
+        for dist, seed in zip(dists, seeds):
+            tfp_dist = dist.init_dist()
+
+            event_shape = tfp_dist.event_shape
+            batch_shape = tfp_dist.batch_shape
+            value_shape = jnp.asarray(dist.at.value).shape  # type: ignore
+            sample_index = len(value_shape) - len(batch_shape) - len(event_shape)
+            sample_shape = value_shape[:sample_index]
+
+            value = tfp_dist.sample(sample_shape, seed)
```

Comment:
I think that here, edge cases could occur: It is not required that users implement the `._sample_n` method of a custom TFP class. In that case, they will receive an error:

```
NotImplementedError: sample_n is not implemented
```

I think this behavior is ok. Do you agree @hriebl ?

Reply:
I would agree that this is the expected behavior. The user can then either implement a sampling method or skip the corresponding node with the `skip` argument.